### ISSUE TITLE
Unsubscribed from channel when component unmounts

### ISSUE THREAD

**Issue by jfulse** [mayteio/use-pusher#43]
**Describe the bug**
Thanks for a great library! If the following issue is as intended please disregard.

When a component with useChannel and useEvent is unmounted, the channel is unsubscribed from although other components are still subscribed. Thus subsequent events are not caught in the still mounted components.

This can be avoided by reorganising the component tree correctly but seems hard to future proof.

**To Reproduce**
I will be happy to make a code sandbox but I'm not sure which Pusher app to connect to etc? Is the idea that I make the sandbox with placeholders and you enter your keys?

**Comment by mayteio** [comment]
Thanks @jfulse, that’s definitely not intended! All subscriptions should continue to receive new events. 

I think this could be solved by introducing a new ChannelProvider - it does the subscription there, passes it down via context and then the useChannel hook consumes the single connection. That would probably solve some other issues here too!

for now, I suggest hoisting your useChannel into a context and providing it high in your app, then using a custom hook to consume this. That way the connection stays live no matter which components get unmourned. 

I’ll write an RFC soon.

**Comment by jfulse** [comment]
Thanks for your response, sounds great!

**Comment by jan-wilhelm** [comment]
just ran into the same issue, this would be greatly appreciated! Thanks for the great lib ;)

**Comment by underfisk** [comment]
@jan-wilhelm Indeed, it does but what i would suggest @mayteio  is that instead of just unsubscribing the channel (checking first if there are any more active subscribers) because this issue was related to having N subscribed and one of them would un-mount and leaves the channel (well this could be good just to help on those cases)

**Comment by devdev-dev** [comment]
Same problem for me. Is there a known workaround to prevent unsubscribing if one of many components unmounts?

**Comment by embryCODE** [comment]
I just ran into this too. Couple of days of back and forth with QA.
